### PR TITLE
ICS-30: updates based on latest implementation

### PR DESCRIPTION
**Pull request by crodriguezvega**
[Rendered view](https://github.com/cosmos/ibc/blob/6bad0a7d390c0d069ec756ab4a87438204b93e2f/spec/app/ics-030-middleware/README.md).

### CODE REVIEW

File: spec/app/ics-030-middleware/README.md [hunk 1]
```diff
@@ -47,9 +47,19 @@ In order to function as IBC Middleware, a module must implement the IBC applicat
...
+Additionally, the middleware must take care to ensure that the application logic can execute its own version negotiation without interference from the nesting middleware. In order to do this, the middleware will format the version in a JSON-encoded string containing the middleware version and the application version (and potentially also other custom parameter fields). The application version may as well be a JSON-encoded string, possibly including further middleware and app versions, if the application stack consists of multiple milddlewares wrapping a base application.  The format of the version string is as follows:
 
-Version: `{middleware_version}:{app_version}`
+```json
+{
+    "<middleware_version_key>": "<middleware_version_value>",
+    "app_version": "<application_version_value>",
+    ... // other custom parameter fields
```
Comment (AdityaSripal) [inline spec/app/ics-030-middleware/README.md:56]:
```suggestion
    // ... other custom parameter fields
```

**Review by AdityaSripal — APPROVED** [review]
Thanks @crodriguezvega !

**Review by mpoke — APPROVED** [review]
LGTM

File: spec/app/ics-030-middleware/README.md [hunk 2]
```diff
@@ -89,101 +104,157 @@ function onChanOpenInit(
   channelIdentifier: Identifier,
   counterpartyPortIdentifier: Identifier,
   counterpartyChannelIdentifier: Identifier,
-  version: string) (version: string, err: Error) {
+  version: string): (version: string, err: Error) {
```
Comment (AdityaSripal) [inline spec/app/ics-030-middleware/README.md:107]:
```suggestion
  version: string) => (version: string, err: Error) {
```

File: spec/app/ics-030-middleware/README.md [hunk 3]
```diff
@@ -89,101 +104,157 @@ function onChanOpenInit(
...
-          cpAppVersion, // note we only pass counterparty app version here
-          appVersion, // only pass app version
-      )
-      abortTransactionUnless(err != nil)
-      versionJSON = {
-          middlewareVersion: middlewareVersion, // note this should have a different field name specific to middleware
-          appVersion: appVersion
-      }
-      return marshalJSON(versionJSON), nil
+  counterpartyVersion: string): (version: string, err: Error) {
```
Comment (AdityaSripal) [inline spec/app/ics-030-middleware/README.md:169]:
```suggestion
  counterpartyVersion: string) => (version: string, err: Error) {
```

File: spec/app/ics-030-middleware/README.md [hunk 4]
```diff
@@ -89,101 +104,157 @@ function onChanOpenInit(
...
+    // by the underlying application, in case it is different than the 
+    // one passed by the caller
+    metadata = {
+        // note this should have a different field name specific to middleware
+        middlewareVersion: metadata.middlewareVersion
+        appVersion: appVersion,
     }
-    return marshalJSON(versionJSON), nil
+
+    return MarshallJSON(version), nil
```
Comment (AdityaSripal) [inline spec/app/ics-030-middleware/README.md:159]:
```suggestion
    return MarshallJSON(metadata), nil
```

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**